### ISSUE TITLE
lsp-ui-sideline pop message break line

### ISSUE THREAD

**Issue by liuchong** [emacs-lsp/lsp-ui#746]
The pop out alert message of lsp-ui-sideline added a line break, when move the cursor on and off the error code, the window will change.
It's annoying, just disable lsp-ui-sideline-mode is ok, maybe. 🔥 

Normal:
<img width="414" alt="image" src="https://user-images.githubusercontent.com/236058/227104681-40e2cc73-d378-45f0-95c4-2f227d7fb8f1.png">

Normal in small window:
<img width="877" alt="image" src="https://user-images.githubusercontent.com/236058/227104809-355cc6a8-c876-4c70-a114-4ca43b048637.png">

Broken in wide window:
<img width="1267" alt="image" src="https://user-images.githubusercontent.com/236058/227104977-f4e2f048-170e-4c7b-87af-645cdd2432f7.png">

Disabled:
<img width="540" alt="image" src="https://user-images.githubusercontent.com/236058/227106206-201b3602-25ed-4f02-bcba-bf1b8c94b779.png">

**Comment by Flinner** [comment]
Is this repeated? #96

**Comment by liuchong** [comment]
> Is this unsolved? #96

I noticed that #96 is an old issue and seems was fixed, not sure if it is same problem as what I found, maybe new version of emacs breaks it again? @Flinner

**Comment by ivan-avalos** [comment]
I'm experiencing this too with Emacs 29.0.91. It makes writing code very annoying, especially with Dart, since Dart has labels at the end of every nested structure, and while typing, lines are constantly jumping as sideline actions appear and disappear.

![Screenshot_20230528_224945](https://github.com/emacs-lsp/lsp-ui/assets/16617314/7f89c856-0965-48d8-a458-539e429e6625)

**Comment by jcs090218** [comment]
I don't know if this will work, but you can try [sideline](https://github.com/emacs-sideline/sideline) and [sideline-lsp](https://github.com/emacs-sideline/sideline-lsp). Pretty much the improved version of `lsp-ui-sideline`.

**Comment by liuchong** [comment]
There seems has similar problem in sideline-lsp, please check the below images:

Normal:
<img width="489" alt="image" src="https://github.com/emacs-lsp/lsp-ui/assets/236058/11d6b084-b296-4979-901f-65a6b93fbd2a">
Sideline enabled:
<img width="890" alt="image" src="https://github.com/emacs-lsp/lsp-ui/assets/236058/a92f883c-7138-4628-8a7d-997ee22ef235">

**Comment by liuchong** [comment]
I think the prompt message on the right should not effect the display of source codes on the left.

**Comment by jcs090218** [comment]
Thanks for testing it out! I will investigate the bug. ;)

**Comment by liuchong** [comment]
@jcs090218 hi, did you fixed it? 👀

**Comment by liuchong** [comment]
Maybe create another layer on top of the editing buffer when popping out the message? @jcs090218 👀

**Comment by jcs090218** [comment]
> Maybe create another layer on top of the editing buffer when popping out the message?

No, we already have `lsp-ui-doc` that does that. I don't think that will solve the issue, but making it customizable might be one solution (use different frontend).
